### DISCUSSION TITLE
VideoClip.write_videofile() is missing with_mask

### DISCUSSION THREAD

**Discussion by amAtgit** [Zulko/moviepy#1823] [Q&A]
while using: moviepy 1.0.3
### requirement:
 I was trying to create a video with alpha channel "rgba" 
### issue:
VideoClip.write_videofile does not have with_mask in its input argument signature. Which means when the scripts hits ffmeg_write_video(), the mask/alpha channel is not stacked on to the frame because withmask is False by default (see code snippet below). I noticed that the pre_release 2.0.0.dev2 has pix_fmt added which helps to set the ffmeg's pix_fmt parameter correctly but this still does not the add the alpha channel to frame in ffmeg_write_video() in ffmpeg_writter.py because there is no way to set withmask to True from VideoClip.write_videofile().

```
        if withmask:
            mask = 255 * clip.mask.get_frame(t)
            if mask.dtype != "uint8":
                mask = mask.astype("uint8")
            frame = np.dstack([frame, mask])

        writer.write_frame(frame)
```

As a result the alpha channel is not added when I do VideoFileClip.write_videofile()
Is this a bug or oversight?
Have I missed something? how do I add alpha channel to a video with moviepy1.0.3 package?

###  possible solution:
I added a with_mask in to VideoClip.write_videofile(with_mask) and then with in the function I passed this flag to ffmpeg_write_video(withmask=with_mask) and this allowed me to create an avi video with alpha channel.

### request:
will the with_mask be added to VideoClip.write_videofile() in near future?
thanks

**Comment by keikoro** [comment]
Please open a proper issue for this (using our issue templates) instead of using GitHub's Discussions feature, thank you.
